src/devices/sound/hc55516.h

```diff
@@ -1,95 +1,212 @@
 // license:BSD-3-Clause
-// copyright-holders:Aaron Giles
+// copyright-holders:Aaron Giles, Jonathan Gevaryahu
+// thanks-to:Zonn Moore
 #ifndef MAME_SOUND_HC55516_H
 #define MAME_SOUND_HC55516_H
 
 #pragma once
 
-class hc55516_device : public device_t, public device_sound_interface
+class cvsd_device : public device_t, public device_sound_interface
 {
 public:
-	hc55516_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	enum
+	{
+		RISING=true,
+		FALLING=false
+	};
+	cvsd_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
```

Comment:
Is this thing a real device at all?  If it’s an abstract base class, it shouldn’t have a public constructor, and the class name should end in _`device_base` to make it clear it’s abstract.

Reply:
This has been renamed to cvsd_device_base and the constructor is now protected rather than public.